### ISSUE TITLE
Enhance/fix: core functionality

### ISSUE THREAD

**Issue by aananthraj** [bitnami-labs/kubewatch#88]
Hi , 

We could improve core functionality of Kubewatch by enhancing/fixing the following functionalities

1.  **Spamming of notifications during start**.
     
    kubewatch bombards the notification channel (say slack) with all the resources that were already present in the cluster. This could be controlled by using `CreationTimestamp` properties available as part of object's metadata.
    
    ```
    // compare CreationTimestamp of objects with Kubewatch serverStartTime
    // and triggers ObjectCreated() only on latest events 
    func (c *Controller) processItem(key string) error {
	  obj, exists, err := c.informer.GetIndexer().GetByKey(key)
      ...
      
      // get object's metedata
      objectMeta := utils.GetObjectMetaData(obj)
      
      ...
      if objectMeta.CreationTimestamp.Sub(serverStartTime).Seconds() > 0 {
	    c.eventHandler.ObjectCreated(obj)
	  }
    }
    ```
    Note: The code change for the above functionality is ready and tested locally. If this fix looks good, we could go ahead raise a PR, review it once and make it live.
    
2. **func ObjectCreated() used for Update and Sync Events.**

     Currently `c.eventHandler.ObjectCreated(obj)` method is being used for the following events
     - Create (valid)
     - Update : invalid. we need to use ObjectUpdated() function instead
     - Sync   : invalid, this could be ignored.
    
     This causes the multiple create notifications during a delete event, since internally a delete event is accompanied with an update and a sync event. we could discuss about this issues and come up a with a solution.
     
     Note: Correct me, if the inference about the delete event is wrong.

Thanks.

**Comment by tylerauerbeck** [comment]
@aananthraj  For topic number 1, I think this is a great idea. I think this was originally brought up in #51 , so if you have a tested solution for this.. go ahead and open a PR and we can look at getting this merged in.

For topic number 2, I delved into this in (probably too much) detail in #80 . I think we definitely need to do a little redevelopment around this part of kubewatch, so I'd be interested in your thoughts on the subject.

**Comment by aananthraj** [comment]
@tylerauerbeck Thanks...!!. I am glad to see your feedback.

Will create the PR for #51. 👍 

> I think we definitely need to do a little redevelopment around this part of kubewatch.

Yes, I was thinking the same. I have few queries with the current implementations. Do we have any Forum (say slack group) to interact elaborately..?

I guess the event handlers implemented using `Skip resync` is not using the recommended client-side caching mechanisms. This could be replaced with `Reflectors`, `Storage` and `DeltaFIFO` etc

**Comment by tylerauerbeck** [comment]
Thanks @aananthraj . I'll try to take a look at this shortly and get it merged in.

As for the questions about the current implementation, I'm not sure that there's an existing slack group for this (although I know there's still a skippbox link on the README.md). Looping in @sebgoa and @ngtuna in case a) you guys are interested in discussing this and b) if there's an existing slack/place where some of these discussions can happen.

If not, we'll find somewhere we can chat about this.

**Comment by aananthraj** [comment]
Created a slack workspace to initiate discussion [kube-watch](https://join.slack.com/t/kube-watch/shared_invite/enQtMzM4MDI1OTg2NjI2LTBjZTI2ODZmYzcwMmIwNTJjMGVkM2NlNjFkNjJkMWFmMDBjOTQxMzMwYmZlZThmYmMzODIwYjNkNzMxNmQ0ZjU). 
we could use it for now and make it official slack workspace later.

**Comment by aananthraj** [comment]
@tylerauerbeck @ngtuna @sebgoa , any updates..?

**Comment by tylerauerbeck** [comment]
Resolved by #99
